core/src/main/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequence.java

```diff
@@ -312,14 +314,15 @@ private MergeCombinePartitioningAction(
       this.cancellationGizmo = cancellationGizmo;
     }
 
+    @SuppressWarnings("unchecked")
```

Comment:
nit: I was looking for where the unchecked operation is used and realized that it is where `TERMINAL` is added to a queue. Could it be better to add this suppression at the line where unchecked operation is actually OK such as adding `TERMINAL` to a queue? The current change seems fine for now but I suppose this warning can do a legit thing if someone modifies some code in this method to use an unchecked operation. This comment is not a blocker.